File: mod/src/client/java/top/fifthlight/armorstand/extension/RenderPipelineSnippetExt.java
package top.fifthlight.armorstand.extension;

import org.jetbrains.annotations.NotNull;
import top.fifthlight.armorstand.model.VertexType;

import java.util.List;
import java.util.Optional;

public interface RenderPipelineSnippetExt {
    @NotNull
    Optional<VertexType> armorstand$getVertexType();

    @NotNull
    Optional<List<String>> armorstand$getUniformBuffers();
}

File: mod/src/client/kotlin/top/fifthlight/armorstand/extension/RenderPipelineSnippetExt.kt
package top.fifthlight.armorstand.extension

import com.mojang.blaze3d.pipeline.RenderPipeline

@Suppress("CAST_NEVER_SUCCEEDS")
val RenderPipeline.Snippet.vertexType
    get() = (this as RenderPipelineSnippetExt).`armorstand$getVertexType`()

@Suppress("CAST_NEVER_SUCCEEDS")
val RenderPipeline.Snippet.uniformBuffers
    get() = (this as RenderPipelineSnippetExt).`armorstand$getUniformBuffers`()
